Task-56988: Photo disappears when it is moved or resized in news editor

AzmiTouil opened the pull request:
Prior this change, when you try to move or resize an uploaded image in the news editor, it disappears.
This is because the editor is trying to get the images after these actions with their uploaded ids, although these uploaded resources have been removed and replaced with the permanent images URLs in the news content when saving the draft.
Proposition: add permanent image URLS in editor when upload is successful

boubaker commented on services/src/main/java/org/exoplatform/news/storage/jcr/JcrNewsStorage.java:
```diff
@@ -250,7 +250,7 @@ public News createNews(News news) throws Exception {
     publicationService.enrollNodeInLifecycle(newsDraftNode, lifecycleName);
     publicationService.changeState(newsDraftNode, PublicationDefaultStates.DRAFT, new HashMap<>());
-    newsDraftNode.setProperty("exo:body", imageProcessor.processImages(news.getBody(), newsDraftNode.getUUID(), "images"));
+    newsDraftNode.setProperty("exo:body", news.getBody());
```
We need to make sure to save Draft with persisted Image to not loose it when we restart the server, since the Uploaded resources will be added into `temp` folder of Tomcat

azayati commented on services/src/main/java/org/exoplatform/news/storage/jcr/JcrNewsStorage.java:
```diff
@@ -613,12 +613,14 @@ public News updateNews(News news, String updater) throws Exception {
     if (newsNode != null) {
-      String processedBody = imageProcessor.processImages(news.getBody(), newsNode.getUUID(), "images");
+      if (PublicationDefaultStates.PUBLISHED.equals(news.getPublicationState())) {
+        newsNode.setProperty("exo:body", imageProcessor.processImages(news.getBody(), newsNode.getUUID(), "images"));
```
IMO you need to set the news body also news.setBody(processedBody);

azayati reviewed (dismissed):
see review

AzmiTouil committed:
Task-56988: Photo disappears when it is moved or resized in news editor
Prior this change, when you try to move or resize an uploaded image in the news editor, it disappears.
This is because the editor is trying to get the images after these actions with their uploaded ids, although these uploaded resources have been removed and replaced with the permanent images URLs in the news content when saving the draft.
Proposition: add permanent image urls in editor when upload is successful

AzmiTouil committed:
Update PR

AzmiTouil committed:
Use extension registry to add select image plugin to the editor

AzmiTouil committed:
PR Review

boubaker commented on webapp/src/main/webapp/news-activity-composer-app/components/ExoNewsActivityComposer.vue:
```diff
@@ -563,17 +566,33 @@ export default {
+          extraPlugins = `${extraPlugins},${ckEditorExtraPlugins}`;
+        }
+        if (ckEditorRemovePlugins) {
+          removePlugins = `${extraPlugins},${ckEditorRemovePlugins}`;
```
```suggestion
          removePlugins = `${removePlugins},${ckEditorRemovePlugins}`;
```

boubaker commented on webapp/src/main/webapp/news-activity-composer-app/components/ExoNewsActivityComposer.vue:
```diff
@@ -563,17 +566,33 @@ export default {
+        if (ckEditorRemovePlugins) {
+          removePlugins = `${removePlugins},${ckEditorRemovePlugins}`;
+        }
+        ckEditorExtensions.map(function (ckEditorExtension) {
```
```suggestion
        ckEditorExtensions.forEach(ckEditorExtension => {
```
You may need to use `forEach` here

boubaker commented on webapp/src/main/webapp/news-activity-composer-app/components/ExoNewsActivityComposer.vue:
```diff
@@ -563,17 +566,33 @@ export default {
+          removePlugins = `${removePlugins},${ckEditorRemovePlugins}`;
+        }
+        ckEditorExtensions.map(function (ckEditorExtension) {
+          newsToolbar.find(toolbarItem => toolbarItem.name === 'links').items.push(ckEditorExtension.extraToolbarItem);
```
shouldn't we do the same for as for `removePlugins` and `extraPlugins` variables ?

AzmiTouil committed:
PR Review
